src/core/border_router/routing_manager.cpp

```diff
@@ -636,7 +636,9 @@ void RoutingManager::StartRouterSolicitationDelay(void)
 {
     uint32_t randomDelay;
 
-    VerifyOrExit(!mRouterSolicitTimer.IsRunning() && mRouterSolicitCount == 0);
+    VerifyOrExit(!IsRouterSolicitationInProgress());
+
+    OT_ASSERT(mRouterSolicitCount == 0);
```

Comment:
@wgtdkp 
I think `mRouterSolicitCount == 0` is always true if `!mRouterSolicitTimer.IsRunning()` ?